cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -363,7 +373,10 @@ class Context(
     val indexStack: SimpleStack<IndexedDataflowGranularity> = SimpleStack(),
     val callStack: SimpleStack<CallExpression> = SimpleStack(),
     var steps: Int,
-) {
+) : HasAssumptions {
+
+    override val assumptions: MutableList<Assumption> = mutableListOf()
```

Comment:
The context is mostly used by the sensitivity classes to check if a step is valid. Can they also modify the assumptions? E.g. if an invokes edge has an assumption? Is there a system to "strip" assumptions again (e.g. if we enter a function and then exit it again, the subsequent nodes may not need that assumption any more if there was no significant impact by the function.

Reply:
No there is no such functionality, if a transition function declares an assumption on the context, all paths containing that node due to the transition, the assumption is relevant for the path.